de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ISarosSession.java

```diff
@@ -387,26 +387,14 @@ public void addSharedResources(
   public void changeColor(int colorID);
 
   /**
-   * FOR INTERNAL USE ONLY !
-   *
-   * <p>Starts queuing of incoming {@linkplain IResourceActivity project-related activities}, since
-   * they cannot be applied before their corresponding project is received and extracted.
+   * Register a activity queuer. Effective only at first call as all following calls are ignored.
    *
-   * <p>That queuing relies on an existing project-to-projectID mapping (see {@link
```

Comment:
Is this comment no longer applicable? It is not mentioned in the interface documentation. If it still contains valuable information, you could use `{@inheritDoc}` to extend the interface javadoc for this method.

Reply:
Calling `addProjectMapping` is relevant for any resource-activity processing, but I'm not sure if this must mentioned explicitly.
I moved it to `ActivityQueuer`, because other impls could use a different approach (like just queue everything).

Reply:
The method states it is for internal use because you should now the internals when calling this. Although I do not know why we added this comment instead of throwing an exception. The mapping is necessary otherwise the marshalled SPath cannot be unmarshalled. The marshalled SPath looks like  i="3234823" p="some/relativePath/foo.txt"

See https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/misc/xstream/SPathConverter.java

Moving the comment to the ActivityQueuer is not helpful, as the activities are not lost there but here

https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java#L709

See also https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/activities/IActivity.java#L84

Reply:
So are you ok with just deleting it?